streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -158,6 +160,22 @@ public StreamThread(TopologyBuilder builder,
                         Metrics metrics,
                         Time time) {
         this(builder, config, null , null, null, jobId, clientId, processId, metrics, time);
+
+        // try to release all the resources on abrupt termination of the thread
+        this.setUncaughtExceptionHandler(new UncaughtExceptionHandler() {
```

Comment:
`super.setUncaughtExceptionHandler` instead of `this.setUncaughtExceptionHandler`?